static/js/user/user.ts

```diff
@@ -19,14 +19,16 @@ import ReactDOM from "react-dom";
 
 import { Page } from "./page";
 
+const dom = "main-section";
+
 window.onload = () => {
-  const info = JSON.parse(document.getElementById("user").dataset.info);
-  const newUser = "True" === document.getElementById("user").dataset.new_user;
+  const user = JSON.parse(document.getElementById(dom).dataset.user);
```

Comment:
document.getElementById is a somewhat expensive operation which we should avoid calling repeatedly if possible. Could we instead hold a reference to the dataset object?